From 45817ad1a3debbe7efb0f17b0e0c979e58485c45 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sun, 10 Dec 2023 12:37:58 +0200
Subject: [PATCH] tests: Clear STA state at the end of ap_ft_reassoc_proto

Do not leave dev[0] in state where it is waiting for
NL80211_CMD_ASSOCIATE to complete since that might deliver an
ASSOC_TIMED_OUT event to the next test case.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 tests/hwsim/test_ap_ft.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/hwsim/test_ap_ft.py b/tests/hwsim/test_ap_ft.py
index 0e9d04775..3980b4c84 100644
--- a/tests/hwsim/test_ap_ft.py
+++ b/tests/hwsim/test_ap_ft.py
@@ -3209,6 +3209,12 @@ def test_ap_ft_reassoc_proto(dev, apdev):
     for t in tests:
         hapd2ap.request("MGMT_RX_PROCESS freq=2412 datarate=0 ssi_signal=-30 frame=" + hdr + ies1 + t)
 
+    # Do not leave dev[0] in state where it is waiting for
+    # NL80211_CMD_ASSOCIATE to complete since that might deliver
+    # an ASSOC_TIMED_OUT event to the next test case.
+    dev[0].request("DISCONNECT")
+    time.sleep(0.2)
+
 def test_ap_ft_reassoc_local_fail(dev, apdev):
     """WPA2-PSK-FT AP Reassociation Request frame and local failure"""
     ssid = "test-ft"
-- 
2.43.0

